### PR TITLE
Altered the self issue

### LINKED ISSUE #1: `find_all_rings` should take a universe as input, not self
the function `find_all_rings` currently has an argument that is not used, modified or returned elsewhere: `self` 
this is probably a copy-paste error.

### PR DESCRIPTION
**Pull request by TristanSJones**
Fixes #1 

Changes made in this Pull Request:
 - Changed the word self to ring_atoms in which ring_atoms is one atom group with all the atoms that have the potential to be atoms within rings in the given universe/all the atoms that will be rings that we want to stack

### CODE REVIEW

**Comment by TristanSJones** [comment]
Fixes self issue described in #1. In addition, it adds tests for ring_distance, ring_normal, find_pi_stacking and find_pi_stacking_rings functions. As well as updated armadillo.py to resolve bugs that were discovered in the test process.

File: armadillo/tests/test_find_pi_stacking_rings.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import pytest
+import sys
+import MDAnalysis as mda
+
+
+def test_armadillo_imported():
+    """Sample test, will always pass so long as import statement worked"""
+    assert "armadillo" in sys.modules
+
+u = mda.Universe('g3.pdb', guess_bonds=True)
```
Comment (micaela-matta) [inline armadillo/tests/test_find_pi_stacking_rings.py:12]: lines 12-17 should all be inside the `test_find_pi_stacking_rings` function

File: armadillo/tests/test_find_all_rings.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import pytest
+import sys
+import MDAnalysis as mda
+
+
+def test_armadillo_imported():
+    """Sample test, will always pass so long as import statement worked"""
+    assert "armadillo" in sys.modules
+
+u = mda.Universe('g3.pdb', guess_bonds=True)
```
Comment (micaela-matta) [inline armadillo/tests/test_find_all_rings.py:12]: line 12 should be inside the `test_find_all_rings` function; the docstring should be immediately after the `def` line.

File: armadillo/tests/test_normal_vector.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import pytest
+import sys
+import MDAnalysis as mda
+
+
+def test_armadillo_imported():
+    """Sample test, will always pass so long as import statement worked"""
+    assert "armadillo" in sys.modules
+
+u = mda.Universe('g3.pdb', guess_bonds=True)
```
Comment (micaela-matta) [inline armadillo/tests/test_normal_vector.py:12]: same as for the other tests, every line/variable definition needed to run the test should be inside the test function

File: armadillo/tests/test_pi_stacking-individual.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# Import package, test suite, and other packages as needed
+import armadillo
+import pytest
+import sys
+import MDAnalysis as mda
+
+
+def test_armadillo_imported():
```
Comment (micaela-matta) [inline armadillo/tests/test_pi_stacking-individual.py:8]: consider whether incorporating all tests into a single  `tests.py` file would make life easier :)

File: armadillo/tests/test_pi_stacking-individual.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    """Sample test, will always pass so long as import statement worked"""
+    assert "armadillo" in sys.modules
+
+u = mda.Universe('g3.pdb', guess_bonds=True)
+
+ring_list = armadillo.find_all_rings(u)
```
Comment (micaela-matta) [inline armadillo/tests/test_pi_stacking-individual.py:14]: if you had a single `test.py` file, you could perhaps load the test `Universe` once, then pass it to all test functions.

File: armadillo/armadillo.py [hunk 6]
```diff
@@ -16,14 +16,31 @@
...
-def find_all_rings(self):
+def find_all_rings(u):
+    """
+    Finds all the rings within a given universe
+    Each ring found is a atom group and so a list of rings is created
+
+    Parameters
+    -----------
+    u :
+        Universie is a MDAnalysis kit that
```
Comment (micaela-matta) [inline armadillo/armadillo.py:37]: seen a few typos here and there:)

File: armadillo/armadillo.py [hunk 7]
```diff
@@ -49,26 +80,89 @@ def normal_vector(ring):
...
+    calc_bonds function with MDAnalysis. The box used is set as u.dimensions
+
+    Parameters
+    -----------
+    ring_a :
+        Atom group containing the atoms of one ring
+    ring_b :
+        Atom group containing the atoms of one ring
+    u :
+        Universie is a MDAnalysis kit that
```
Comment (micaela-matta) [inline armadillo/armadillo.py:95]: would suggest replacing with something like: `MDAnalysis Universe containing topology and trajectory info`

File: armadillo/armadillo.py [hunk 8]
```diff
@@ -94,6 +191,15 @@ def find_pi_stacking_rings(list_of_rings, distmax, distmin,
     --------
     Returns a lists with n = [0] being ix_of_actually_pi_stacking and n= [1] ix_of_none_pi_stacking
     '''
+
+    ring_cogs = []
+    for ring in ring_list:
+        ring_cogs.append(ring.center_of_geometry())
```
Comment (micaela-matta) [inline armadillo/armadillo.py:197]: why don't you use`ring_center`?
you could try list comprehension instead of a loop:)

**Review by micaela-matta — CHANGES_REQUESTED** [review]
hi @TristanSJones, have a look at the comments and let me know if anything isn't clear! Great job towards your first PR 🥇

**Pull request merged**